OrbitGl/EventTrack.cpp

```diff
@@ -187,15 +187,53 @@ bool EventTrack::IsEmpty() const {
 }
 
 //-----------------------------------------------------------------------------
-std::string EventTrack::GetSampleTooltip(PickingID id) const {
-  auto SafeGetFormattedFunctionName = [](uint64_t addr) -> std::string {
-    auto it = Capture::GAddressToFunctionName.find(addr);
-    return it == Capture::GAddressToFunctionName.end()
-               ? std::string("<i>") + "???" + "</i>"
-               : it->second;
-  };
+std::string SafeGetFormattedFunctionName(uint64_t addr, int max_line_length) {
+  auto it = Capture::GAddressToFunctionName.find(addr);
+  if (it == Capture::GAddressToFunctionName.end()) {
+    return std::string("<i>") + "???" + "</i>";
+  }
 
-  static const std::string unknown_return_text = "Function call information missing";
+  std::string fn_name = ShortenStringWithEllipsis(it->second, max_line_length);
+  // Simple HTML escaping
+  fn_name = Replace(fn_name, "&", "&amp;");
+  fn_name = Replace(fn_name, "<", "&lt;");
+  fn_name = Replace(fn_name, ">", "&gt;");
+  return fn_name;
+};
+
+//-----------------------------------------------------------------------------
+std::string FormatCallstackForTooltip(std::shared_ptr<CallStack> callstack,
+                                      int max_line_length = 80,
+                                      int max_lines = 13) {
```

Comment:
This seems to be quite a limitation. What if the user prefers to see the entire callstack, even if that might take a lot of screen space?

Reply:
You always have the option to select the sample and see the full callstack in the "selection" tab. TBH, probably the same limitation may be too much for some devs and too little for others...
Alternatively, we could start introducing UI settings for things like those, but it seems a bit of an overkill to add this for a relatively small feature like this?

Reply:
Some limitation is obviously needed. As @reichlfl points out, users can select the sample and see the callstack then on the right. Since this is mentioned in the tooltip, it should be somewhat discoverable / understandable for users and I think that's good enough for now. Eventually, we can add a button ("More", "See entire callstack",  "Select this sample") to the tooltip (though I don't know how much work that would be). In any case, we should aim to solve this directly in the UI, rather than introduce settings that may themselves be confusing for users.

Reply:
I know that samples can be selected and seen in the "selection" tab, but that's *very* tedious to see a single callstack and one reason for the tooltips. But of course a UI setting for this is definitely overkill. 
So let me say it differently: 13 frames (or 12?) seem to be quite on the small side. It basically means the last six as the frames that get hidden are the ones in the middle. From what I've seen such a number of lines still produces quite a small tooltip. I would prefer to see more information (with some limitation obviously) than a small cute tooltip that can't be enlarged.
(Of course this is very easy to change in the future but again it's one of those things that once there don't really get changed anymore.)

Reply:
I was just pointing out that some limitation is always needed (above you were talking about users wanting to see the entire callstack) and that there is a workaround to see the full callstack. Agree that we may want to increase 13 to some larger number, maybe just try 25 to double the number of frames visible? 

We might also want to think about doing this asymmetrically where users can see more frames at the top of the stack (the side including the current function). Though not always true, that seems to be in general more relevant. WDYT?

But in general, you will always have a case where someone wants/needs to see the entire stack. We just want to make that case as rare as possible while maintaining good usability of the tooltips.

Reply:
@ronaldfw I also discussed this offline with @reichlfl and I think that we all agree on what you said now.

Reply:
I settled for 20 now, this looks ok on my smaller (1080p) screen - 13 may have been a bit short, true.
@ronaldfw Actually it already is asymmetrical, exactly as you suggest: When the stack is shortened, it only shows a fixed number of lines from the bottom, then "...", then as much as possible from the top. I added the number of lines to display from the bottom as a parameter to make it clearer and more verbose.

Reply:
Sounds good!